src/planning.ts

```diff
@@ -476,109 +520,115 @@ function dedupPoints(points: Vec2[], epsilon: number): Vec2[] {
-    const distance = segment.length();
-    const vInitial = segment.entryVelocity;
-    const vExit = nextSegment.maxEntryVelocity;
-    const p1 = segment.p1;
-    const p2 = segment.p2;
-
-    const m = computeTriangle(distance, vInitial, vExit, accel, p1, p2);
+    const segment = segments[i]
+    const nextSegment = segments[i + 1]
+    const distance = segment.length()
+    const vInitial = segment.entryVelocity
+    const vExit = nextSegment.maxEntryVelocity
+    const p1 = segment.p1
+    const p2 = segment.p2
+
+    const m = computeTriangle(distance, vInitial, vExit, accel, p1, p2)
     if (m.s1 < -epsilon) {
       // We'd have to start decelerating _before we started on this segment_. backtrack.
       // In order enter this segment slow enough to be leaving it at vExit, we need to
       // compute a maximum entry velocity s.t. we can slow down in the distance we have.
       // TODO: verify this equation.
-      segment.maxEntryVelocity = Math.sqrt(vExit * vExit + 2 * accel * distance);
-      i -= 1;
+      segment.maxEntryVelocity = Math.sqrt(vExit * vExit + 2 * accel * distance)
+      i -= 1
     } else if (m.s2 <= 0) {
       // No deceleration.
       // TODO: shouldn't we check vMax here and maybe do trapezoid? should the next case below come first?
-      const vFinal = Math.sqrt(vInitial * vInitial + 2 * accel * distance);
-      const t = (vFinal - vInitial) / accel;
+      const vFinal = Math.sqrt(vInitial * vInitial + 2 * accel * distance)
+      const t = (vFinal - vInitial) / accel
       segment.blocks = [
         new Block(accel, t, vInitial, p1, p2)
-      ];
-      nextSegment.entryVelocity = vFinal;
-      i += 1;
+      ]
+      nextSegment.entryVelocity = vFinal
+      i += 1
     } else if (m.vMax > vMax) {
       // Triangle profile would exceed maximum velocity, so top out at vMax.
-      const z = computeTrapezoid(distance, vInitial, vMax, vExit, accel, p1, p2);
+      const z = computeTrapezoid(distance, vInitial, vMax, vExit, accel, p1, p2)
       segment.blocks = [
         new Block(accel, z.t1, vInitial, z.p1, z.p2),
         new Block(0, z.t2, vMax, z.p2, z.p3),
         new Block(-accel, z.t3, vMax, z.p3, z.p4)
-      ];
-      nextSegment.entryVelocity = vExit;
-      i += 1;
+      ]
+      nextSegment.entryVelocity = vExit
+      i += 1
     } else {
       // Accelerate, then decelerate.
       segment.blocks = [
         new Block(accel, m.t1, vInitial, m.p1, m.p2),
         new Block(-accel, m.t2, m.vMax, m.p2, m.p3)
-      ];
-      nextSegment.entryVelocity = vExit;
-      i += 1;
+      ]
+      nextSegment.entryVelocity = vExit
+      i += 1
     }
   }
-  const blocks: Block[] = [];
+  const blocks: Block[] = []
   segments.forEach((s) => {
     s.blocks.forEach((b) => {
       if (b.duration > epsilon) {
-        blocks.push(b);
+        blocks.push(b)
       }
-    });
-  });
-  return new XYMotion(blocks);
+    })
+  })
+  return new XYMotion(blocks)
 }
```

Comment:
Semicolon noise snuck in, I'm hoping after this merge to do `npm run format` and have that be a separate PR